tests/assembly/targets/targets-macho.rs

```diff
@@ -27,6 +27,12 @@
 //@ revisions: arm64_32_apple_watchos
 //@ [arm64_32_apple_watchos] compile-flags: --target arm64_32-apple-watchos
 //@ [arm64_32_apple_watchos] needs-llvm-components: aarch64
+//@ revisions: aarch64_apple_visionos
+//@ [aarch64_apple_visionos] compile-flags: --target aarch64-apple-visionos
+//@ [aarch64_apple_visionos] needs-llvm-components: aarch64
+//@ revisions: aarch64_apple_visionos_sim
+//@ [aarch64_apple_visionos_sim] compile-flags: --target aarch64-apple-visionos-sim
+//@ [aarch64_apple_visionos_sim] needs-llvm-components: aarch64
```

Comment:
Unfortunately, I think these tests need to be commented out for now since LLVM 16 doesn't know what these targets are which is causing that builder to fail. 

See for example:

https://github.com/rust-lang/rust/blob/3cbb93223f33024db464a4df27a13c7cce870173/tests/assembly/targets/targets-elf.rs#L180C1-L188C3

Reply:
Now that you mention it, I see the LLVM 16 test. Do you know why it's being tested against an old version of LLVM?

Reply:
We support bringing your own version of LLVM rather than our version which is mostly used by Linux distros so that LLVM uses the same LLVM that they ship as a package. To enable that to work better, we support current stable LLVM as well as a few older releases as often distros don't support the most recent stable LLVM as quickly as we do. 

LLVM 16 is currently the oldest LLVM we support, so there is a CI job that compiles with it. Ideally given that LLVM 18 is required for these targets, we'd provide a nice error message if you try to use them with an older LLVM but, at the same time, this is a Tier 3 target so the level of support we promise just isn't [very high](https://doc.rust-lang.org/rustc/target-tier-policy.html#tier-3-target-policy).

Reply:
I believe `//@ [aarch64_apple_visionos_sim] min-llvm-version: 18` should work -- no need to comment them out entirely.

Reply:
Oh, awesome! I was trying to figure out how to use `min-llvm-version` here. Thanks @nikic!